apps/external_thread_pool_demo/micro_system.cc

```diff
@@ -0,0 +1,61 @@
+/**
+ * DAIL in Alibaba Group
+ *
+ */
+
+#include "micro_system.hh"
+#include "my_thread_pool.hh"
+
+void micro_system::register_pollers() {
+    _my_thread_pool_poller = seastar::reactor::poller(std::make_unique<my_thread_pool_pollfn>());
+}
+
+void micro_system::deregister_pollers() {
+    _my_thread_pool_poller = {};
+}
+
+void micro_system::clear_micro_system() {
+    micro_engine().deregister_pollers();
```

Comment:
Why do you need to deregister pollers on this shard first before doing that on all others?
Won't `invoke_on_all` work?
Also, I don't understand the logic below.  The semaphore doesn't help with anything,
and you run the async fiber in the background and don't wait for it.

It'd be best if you return a future<> here and do something like:
```c++
    return seastar::smp::invoke_on_all([] (unsigned) {
        micro_engine().deregister_pollers();
    }).then([] {
        my_thread_pool::stop();
    });
}

Then, I'd separate the call to `clear_micro_system` into a future-returning `stop()` methods, that neededs to be called, and waited on, before exit(), and leave only the exit() function as synchronous.
```

Reply:
In our own system, we need to do something on main shard before other shards when clearing our system, but in order to simply show this demo, we have deleted this part of the content, so if it is only in this example, we can completely follow what you said Do it!

We refer to seastar::reactor::stop() for the usage of semaphore, maybe your suggestion is a better approach, I can try to refactor it.

Reply:
I just revised a version, but I'm not sure if this is what you expect, or if it looks like the following：

`
void micro_system::exit() {
        seastar::engine().exit(0);
}

// in demo_app_example.cc
...
}).then([] {
    return micro_engine().stop();
}).then([] {
    micro_engine().exit();
});
...
`

Reply:
yes, that's the way I envisioned it, thanks.